idaes/models_extra/power_generation/unit_models/cpu.py

```diff
@@ -0,0 +1,808 @@
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+##############################################################################
+# The development of this flowsheet/code is funded by the ARPA-E DIFFERENTIATE
+# project: “Machine Learning for Natural Gas to Electric Power System Design”
+# Project number: DE-FOA-0002107-1625.
+# This project is a collaborative effort between the Pacific Northwest National
+# Laboratory, the National Energy Technology Laboratory, and the University of
+# Washington to design NGFC systems with high efficiencies and low CO2
+# emissions.
+##############################################################################
+"""
+IDAES block for surrogate models of a CO2 purifucation process
+
+This is a black box (surrogate based) model for CO2 purification process
+designed in Aspen.
+
+The unit has 1 inlet stream for CO2-rich flue gas: inlet
+The unit has 3 outlet streams: pure_CO2, water, and vent
+
+The state variables are flow_mol, mol_fraction, temperature, and pressure.
+The state variables can be accessed via ports named: inlet, pure_CO2, water,
+and vent_gas
+
+The surrogate models were prepared by varying the inlet flowrate and
+composition. The expected value is 11350 lbmol/hr, lower bound: 10850 lbmol/hr,
+and upper bound: 11850 lbmol/hr.
+300 samples have been developed using Latin Hypercube Sampling
+Method, the data set is trained by running the CPU rigorous model. ALAMO was
+used to fit the surrogate models.
+Outputs are a function of the inlet flowrate and species composition.
+Inlet temperature and pressure are assumed to be fixed at 100 F and 14.7 psia.
+The degrees of freedom are the inlet states:
+(1) Inlet flow in mol/s
+(2) Inlet component mol fractions: CO2, H2O, N2, Ar, O2
+"""
+
+# Import IDAES cores
+from idaes.core import declare_process_block_class, UnitModelBlockData
+import idaes.core.util.scaling as iscale
+# Additional import for the unit operation
+import pyomo.environ as pyo
+from pyomo.environ import Var, Constraint, units as pyunits
+from pyomo.network import Port
+import idaes.logger as idaeslog
+
+__author__ = "Differentiate Team (N. Susarla, A. Noring, M. Zamarripa)"
+__version__ = "1.1.0"
+
+
+# ----------------------------------------------------------------------------
+@declare_process_block_class("CPU")
+class CPUData(UnitModelBlockData):
```

Comment:
The long-form `CarbonProcessingUnit` should be used instead of `CPU` when defining a model like this.